### DISCUSSION TITLE
How to expose assets in UI library?

### DISCUSSION THREAD

**Discussion by AlanSanchezP** [nrwl/nx#14258] [General]
I'm currently working on a react monorepo for my company with the goal of extracting reusable code that multiple teams can use in their projects. The first package under development is a set of UI components and assets (mainly svg files).

Is there a way to expose an `assets/` folder in the package.json generated by nx?
I'm currently trying to add this exports map:
```
"exports": {
    "./assets/*": "./assets/*",
    ".": {
        "import": "./index.mjs",
        "require": "./index.js"
    }
}
```

however, it doesn't seem to work, as I get the following error when trying to import an asset from a test app package inside the same monorepo:

`[vite] Internal server error: Failed to resolve import "@org/ui/assets/TestAsset.svg" from "packages/demo-app/src/app/app.tsx". Does the file exist?`

I'm not sure what's the error here, as if I try to expose the same index file under a different path, I get the same error (/test not being found as a valid import on the test app):
```
"exports": {
    "./test": {
      "import": "./index.mjs",
      "require": "./index.js"
    },
    ".": {
      "import": "./index.mjs",
      "require": "./index.js"
    }
}
```

Is there a way to extend the exports map, and if there is, can I expose an asset folder this way? 

An important thing to note is that the test app package is only being used for development, but the ui package (and future ones to exist inside this monorepo) is meant to be published to a private registry.

**Comment by achinchen** [comment]
I think you can give this package [vite-plugin-lib-inject-css](https://www.npmjs.com/package/vite-plugin-lib-inject-css) a try.
Reference: https://github.com/vitejs/vite/issues/1579#issuecomment-1483756199
